src/HonzaBotner.Discord.Services/Commands/FunCommands.cs

```diff
@@ -1,58 +1,55 @@
-using DSharpPlus.Entities;
-using Emzi0767;
+using DSharpPlus.SlashCommands;
 using HonzaBotner.Discord.Services.Extensions;
-using Microsoft.Extensions.Logging;
 
 namespace HonzaBotner.Discord.Services.Commands;
 
-[Group("fun")]
-[Description("Commands that bring fun!")]
-[RequireGuild]
-public class FunCommands : BaseCommandModule
+public class FunCommands : ApplicationCommandModule
 {
-    private readonly ILogger<FunCommands> _logger;
 
-    public FunCommands(ILogger<FunCommands> logger)
+    [SlashCommand("choose", "Pick one of provided options")]
+    public async Task ChooseCommandAsync(
+        InteractionContext ctx,
+        [Option("options", "Options to choose from. Default delimiter: ,")]
+        string options,
+        [Option("delimiter", "Character that separates options. Default: \",\"")]
+        [MinimumLength(1), MaximumLength(1)]
+        string delimiter = ","
+        )
     {
-        _logger = logger;
-    }
-
-    [Command("choose")]
-    [Aliases("decide", "pick")]
-    [Description("Decide a destiny.")]
-    public async Task Choose(
-        CommandContext ctx,
-        [Description("Choices to decide from.")] [RemainingText]
-        params string[] choices
-    )
-    {
-        await ctx.TriggerTypingAsync();
-
-        SecureRandom random = new();
-        string selected = choices[random.Next(choices.Length)].RemoveDiscordMentions(ctx.Guild);
-
-        try
+        var answers = options.Split(delimiter, StringSplitOptions.TrimEntries & StringSplitOptions.RemoveEmptyEntries)
+            .Select(option => option.Trim().RemoveDiscordMentions(ctx.Guild))
+            .Where(option => option != "").ToArray();
+        if (answers.Length == 0)
         {
-            if (selected.Trim().Length == 0)
-            {
-                await ctx.Message.CreateReactionAsync(DiscordEmoji.FromName(ctx.Client, ":-1:"));
-            }
-            else
+            await ctx.CreateResponseAsync("Nope");
+            return;
+        }
+        Random random = new();
+        var text = new StringBuilder("I picked: ");
+        var winNumber = random.Next(answers.Length);
+        var winner = answers[winNumber];
+        text.Append("`" + winner + "`");
+        if (answers.Length > 1)
+        {
+            text.Append("\nOptions were:\n");
+            foreach (var option in answers)
             {
-                await ctx.Channel.SendMessageAsync(
-                    (choices.Length == 1 ? "Jak prosté: " : "Vybral jsem: ")
-                    + $"**{selected}**"
-                );
+                text.Append("`" + option+ "`, ");
             }
+
+            text.Remove(text.Length - 2, 2);
         }
-        catch (Exception e)
+        else
         {
-            _logger.LogWarning(e, "Couldn't send a message");
-            await ctx.Message.CreateReactionAsync(DiscordEmoji.FromName(ctx.Client, ":-1:"));
+            text.Append("\nAre you bipolar or why was this necessary?!");
```

Comment:
?

Reply:
Will add more random text in the future. Idea is to question someone's ability to think clearly when he offers only one thing to choose from. Might be funny. Might not if Kalvoda uses it....